File: Legacy/ERP.Entities/Brand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ERP.Entities
{
    public class Brand : BaseClass
    {
        public int BraId { get; set; }
        public int SocId { get; set; }
        public string BraCode { get; set; }
        public string BraName { get; set; }
        public string BraDescription { get; set; }
        public bool BraIsActived { get; set; }
    }
}

File: frontend/src/types/brand.ts
/**
 * Brand entity representing a product brand in the ERP system
 */
export interface Brand {
  braId: number
  socId: number
  braCode: string
  braName: string
  braDescription?: string
  braIsActived: boolean
  fId?: string
}

/**
 * DTO for creating a new brand
 */
export interface BrandCreateDto {
  braCode: string
  braName: string
  braDescription?: string
  braIsActived?: boolean
}

/**
 * DTO for updating an existing brand
 */
export interface BrandUpdateDto extends Partial<BrandCreateDto> {
  braId: number
}

/**
 * Search/filter parameters for brand list
 */
export interface BrandSearchParams {
  search?: string
  isActive?: boolean
  page?: number
  pageSize?: number
  sortBy?: string
  sortOrder?: 'asc' | 'desc'
}

/**
 * Brand list item (summary view)
 */
export interface BrandListItem {
  braId: number
  braCode: string
  braName: string
  braDescription?: string
  braIsActived: boolean
}
